docs/SONATA_DEVELOPER_GUIDE.md

```diff
@@ -904,107 +904,98 @@ The indexing is rooted at a single group called indices which is a subgroup of a
 
-        "$BASE_DIR": "/path/to/my/workspace",
-
-        "$NETWORK_DIR": "$BASE_DIR/networks",
-
-        "$COMPONENT_DIR": "$BASE_DIR/components"
-
-    },
+Example:
+```
+"target_simulator": "NEURON"
+```
 
-    "components_dir": {
+#### manifest
 
-The directory in which to find the neuronal morphology files:
+Provides a convenient handle on setting variables that point to base paths. These variables can be then used in the rest of the config file to point to various directories that share the first portion of the path. One can also use '.' in the beginning of any path to declare a path relevant to the network config file location.
 
-        "morphologies_dir": "$COMPONENT_DIR/morphologies",
+Example:
+```
+"manifest": {
+  "$BASE_DIR": "./",
+  "$NETWORK_DIR": "$BASE_DIR/networks",
+  "$COMPONENT_DIR": "$BASE_DIR/components"
+}
+```
 
-The directory in which to find the edge_types and point neuron node_types dynamics_params .json files, respectively:
+#### components_dir
 
-        "synaptic_models_dir": "$COMPONENT_DIR/synapse_dynamics",
+Base directories for different components, in particular:
 
-        "point_neuron_models_dir": "$COMPONENT_DIR/point_neuron_dynamics",
+| key | content |
+|-----|---------|
+| `morphologies_dir` | Neuronal morphology SWC files |
+| `synaptic_models_dir` | Edge types `dynamics_params` JSON files |
+| `point_neuron_models_dir` | Point neuron node_types `dynamics_params` JSON files |
+| `mechanisms_dir` | Pre-compiled mechanisms (optional, relevant for NEURON networks) |
+| `biophysical_neuron_models_dir` | Biophysical neuron model types NeuroML |
+| `templates` | HOC templates for the edges |
 
-Mechanisms may need to be compiled in advance, depending on implementation.  This field is optional, and is relevant for NEURON networks:
 
-        "mechanisms_dir":"$COMPONENT_DIR/mechanisms",
+Example:
 
-Where to find the .nml for biophysical neuron model types:
+```
+"components_dir": {
+  "morphologies_dir": "$COMPONENT_DIR/morphologies",
+  "synaptic_models_dir": "$COMPONENT_DIR/synapse_dynamics",
+  "point_neuron_models_dir": "$COMPONENT_DIR/point_neuron_dynamics",
+  "mechanisms_dir":"$COMPONENT_DIR/mechanisms",
+  "biophysical_neuron_models_dir": "$COMPONENT_DIR/biophysical_neuron_dynamics",
+  "templates": "$COMPONENT_DIR/hoc_templates"
+}
+```
 
-        "biophysical_neuron_models_dir": "$COMPONENT_DIR/biophysical_neuron_dynamics",
+#### networks
 
-Where to find the hoc templates for the edges:
+A network is defined by nodes and edges. In the example below, a V1 model is being simulated (with recurrent connections) that receives input from virtual LGN source nodes. Each population of nodes should contain `nodes` and `node_types` while each population of edges should contain `edges` and `edge_types`. If some node population does not use `node_types` it could be `null` (same for edge population).
```

Comment:
We have already discussed it internally, but it looks really weird making the node_types_file mandatory as well as the node_type_ids to then not user them at all and having to specify how to leave then empty. Wouldn't it be better not making them mandatory?

Reply:
I'm fine with making both optional.